[Bug]: TS7016 error when using an o3r sdk in an app with `moduleResolution: bundler`

### Package name

core

### Package version

12.0.9

### Reproduction steps

This issue occurs with a new Angular generated application:

```sh
ng new app
cd app
npm install @scope/sdk
```

In `apps/cockpit/src/app/app.component.ts`, add the following code:
```ts
// ...
import { ScopeApi } from '@scope/sdk';

// ...
constructor() {
    const scopeApi = new ScopeApi(undefined);
}
```

### Current result

If we try to run `ng serve`, the following error is raised:
```sh
X [ERROR] TS7016: Could not find a declaration file for module '@dapi/sdk'. 'C:/git_clones/test/ng19new/node_modules/@scope/sdk/esm2020/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/scope__sdk` if it exists or add a new declaration (.d.ts) file containing `declare module '@scope/sdk';` [plugin angular-compiler]

    src/app/app.component.ts:3:29:
      3 │ import { ScopeApi } from '@scope/sdk';
        ╵                              ~~~~~~~~~~~
```

### Expected result

No error is raised.

### Additional comments

Note: this is due to the following field in `tsconfig.json`:
```json
"moduleResolution": "bundler",
```
Because switching to `"moduleResolution": "node"` fixes the issue.

However, `bundler` is the value that is defined by default when generating an app with `ng new`, so I am not sure switching to `node` is a proper solution.

This issue seems to be due to the way the sdk is packaged, which is why I am raising it in `@ama-sdk` package (I may by wrong though, as I am not familiar with this process).

--- a/apps/showcase/tsconfig.app.json
+++ b/apps/showcase/tsconfig.app.json
@@ -21,7 +21,7 @@
     "downlevelIteration": true,
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "moduleResolution": "node",
+    "moduleResolution": "bundler",
     "resolveJsonModule": true,
     "importHelpers": true,
     "target": "ES2022",

--- a/packages/@ama-sdk/schematics/migration.json
+++ b/packages/@ama-sdk/schematics/migration.json
@@ -25,6 +25,11 @@
       "version": "11.4.0-alpha.0",
       "description": "Updates of @ama-sdk/schematics to v11.4.*",
       "factory": "./schematics/ng-update/index#updateV11_4"
+    },
+    "migration-v12_1_3": {
+      "version": "12.1.3",
+      "description": "Updates of @ama-sdk/schematics to v12.1.3",
+      "factory": "./schematics/ng-update/index#updateV12_1_3"
     }
   }
 }

--- a/packages/@ama-sdk/schematics/schematics/ng-update/index.ts
+++ b/packages/@ama-sdk/schematics/schematics/ng-update/index.ts
@@ -1,5 +1,5 @@
-import type {
-  Rule,
+import {
+  type Rule,
 } from '@angular-devkit/schematics';
 import {
   isTypescriptSdk,
@@ -10,6 +10,7 @@ import {
   updateV10_3 as tsUpdateV10_3,
   updateV11_0 as tsUpdateV11_0,
   updateV11_4 as tsUpdateV11_4,
+  updateV12_1_3 as tsUpdateV12_1_3,
 } from './typescript';
 
 /**
@@ -76,6 +77,20 @@ export function updateV11_4(): Rule {
   return (tree, context) => {
     if (isTypescriptSdk(tree)) {
       return tsUpdateV11_4()(tree, context);
+    }
+
+    return tree;
+  };
+}
+
+/**
+ * Update of Ama-sdk library V12.1.3
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention -- function name contains the version
+export function updateV12_1_3(): Rule {
+  return (tree, context) => {
+    if (isTypescriptSdk(tree)) {
+      return tsUpdateV12_1_3()(tree, context);
     }
 
     return tree;

--- a/packages/@ama-sdk/schematics/schematics/ng-update/typescript/index.ts
+++ b/packages/@ama-sdk/schematics/schematics/ng-update/typescript/index.ts
@@ -22,6 +22,9 @@ import {
 import {
   createGitAttributesFile,
 } from './v11.4/create-gitattributes';
+import {
+  clearPackageJsonExports,
+} from './v12.1/clean-packagejson-exports';
 
 /**
  * update of Otter library V10.0
@@ -81,6 +84,18 @@ export function updateV11_0(): Rule {
 export function updateV11_4(): Rule {
   const updateRules: Rule[] = [
     createGitAttributesFile
+  ];
+
+  return chain(updateRules);
+}
+
+/**
+ * Update of Ama-sdk library V12.1.3
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention -- function name contains the version
+export function updateV12_1_3(): Rule {
+  const updateRules: Rule[] = [
+    clearPackageJsonExports
   ];
 
   return chain(updateRules);

--- a/packages/@ama-sdk/schematics/schematics/ng-update/typescript/v12.1/clean-packagejson-exports.spec.ts
+++ b/packages/@ama-sdk/schematics/schematics/ng-update/typescript/v12.1/clean-packagejson-exports.spec.ts
@@ -1,0 +1,26 @@
+import {
+  readFileSync,
+} from 'node:fs';
+import {
+  resolve,
+} from 'node:path';
+import {
+  Tree,
+} from '@angular-devkit/schematics';
+import {
+  clearPackageJsonExports,
+} from './clean-packagejson-exports';
+
+const mockPath = resolve(__dirname, '..', '..', '..', '..', 'testing', 'mocks', 'updates', 'v12');
+
+describe('clearPackageJsonExports', () => {
+  test('should edit correctly previously generated SDK', async () => {
+    const tree = Tree.empty();
+    tree.create('package.json', readFileSync(resolve(mockPath, 'shell-package.json'), { encoding: 'utf8' }));
+    await clearPackageJsonExports(tree, {} as any);
+
+    const content = tree.readText('package.json');
+    expect(content)
+      .toEqual(readFileSync(resolve(mockPath, 'shell-package.results.json'), { encoding: 'utf8' }));
+  });
+});

--- a/packages/@ama-sdk/schematics/schematics/ng-update/typescript/v12.1/clean-packagejson-exports.ts
+++ b/packages/@ama-sdk/schematics/schematics/ng-update/typescript/v12.1/clean-packagejson-exports.ts
@@ -1,0 +1,57 @@
+import type {
+  Rule,
+} from '@angular-devkit/schematics';
+import {
+  findFilesInTree,
+} from '@o3r/schematics';
+import {
+  type PackageJson,
+} from 'type-fest';
+
+/**
+ * Update the package.json files generated in the Shell of the typescript SDK to support `bundler`
+ * @param tree
+ */
+export const clearPackageJsonExports: Rule = (tree) => {
+  const files = findFilesInTree(tree.root, (p) => p.endsWith('package.json'));
+
+  files.forEach((file) => {
+    const packageJson = tree.readJson(file.path) as PackageJson;
+
+    packageJson.types = packageJson.types || packageJson.typings;
+
+    delete packageJson.es2020;
+    delete packageJson.esm2020;
+    delete packageJson.es2015;
+    delete packageJson.esm2015;
+    delete packageJson.typings;
+
+    if (packageJson.exports) {
+      packageJson.exports = Object.fromEntries(
+        Object.entries(packageJson.exports)
+          .map(([path, map]) => {
+            if (typeof map !== 'object' || Array.isArray(map) || map === null) {
+              return [path, map];
+            }
+
+            if (map.typings) {
+              map.types = map.typings;
+            }
+            if (!map.module) {
+              map.module = map.esm2020 || map.esm2015;
+            }
+            delete map.es2020;
+            delete map.esm2020;
+            delete map.es2015;
+            delete map.esm2015;
+            delete map.typings;
+
+            return [path, map];
+          })
+      );
+    }
+
+    tree.overwrite(file.path, JSON.stringify(packageJson, null, 2) + '\n');
+  });
+  return tree;
+};

--- a/packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/package.json.template
+++ b/packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/package.json.template
@@ -6,11 +6,7 @@
   "main": "cjs/index.js",
   "module": "esm2020/index.js",
   "node": "cjs/index.js",
-  "es2020": "esm2020/index.js",
-  "esm2020": "esm2020/index.js",
-  "es2015": "esm2015/index.js",
-  "esm2015": "esm2015/index.js",
-  "types": "index.d.ts",
+  "types": "./index.d.ts",
   "sideEffects": false,
   "exports": {
     "./package.json": {
@@ -19,13 +15,11 @@
     ".": {
       "main": "./cjs/index.js",
       "module": "./esm2020/index.js",
-      "es2020": "./esm2020/index.js",
-      "esm2020": "./esm2020/index.js",
       "node": "./cjs/index.js",
-      "default": "./esm2020/index.js",
-      "typings": "./index.d.ts",
+      "types": "./index.d.ts",
       "import": "./esm2020/index.js",
-      "require": "./esm2020/index.js"
+      "require": "./esm2020/index.js",
+      "default": "./esm2020/index.js"
     }
   },
   "scripts": {

--- a/packages/@ama-sdk/schematics/testing/mocks/updates/v12/shell-package.json
+++ b/packages/@ama-sdk/schematics/testing/mocks/updates/v12/shell-package.json
@@ -1,0 +1,60 @@
+{
+  "name": "@o3r-training/showcase-sdk",
+  "version": "0.0.0-placeholder",
+  "description": "showcase-sdk - SDK",
+  "source": "src/index.ts",
+  "main": "cjs/index.js",
+  "module": "esm2020/index.js",
+  "node": "cjs/index.js",
+  "es2020": "esm2020/index.js",
+  "esm2020": "esm2020/index.js",
+  "es2015": "esm2015/index.js",
+  "esm2015": "esm2015/index.js",
+  "types": "index.d.ts",
+  "sideEffects": false,
+  "deprecated": "This package is intended for testing purposes only.",
+  "publishConfig": {
+    "access": "public"
+  },
+  "exports": {
+    "./package.json": {
+      "default": "./package.json"
+    },
+    ".": {
+      "main": "./cjs/index.js",
+      "module": "./esm2020/index.js",
+      "es2020": "./esm2020/index.js",
+      "esm2020": "./esm2020/index.js",
+      "node": "./cjs/index.js",
+      "default": "./esm2020/index.js",
+      "typings": "./index.d.ts",
+      "import": "./esm2020/index.js",
+      "require": "./esm2020/index.js"
+    },
+    "./open-api.yaml": "./open-api.yaml",
+    "./helpers": {
+      "main": "./cjs/helpers/index.js",
+      "es2020": "./esm2020/helpers/index.js",
+      "esm2020": "./esm2020/helpers/index.js",
+      "node": "./cjs/helpers/index.js",
+      "default": "./esm2020/helpers/index.js",
+      "typings": "./helpers/index.d.ts",
+      "import": "./esm2020/helpers/index.js",
+      "require": "./esm2020/helpers/index.js"
+    }
+  },
+  "dependencies": {
+    "@swc/helpers": "~0.5.0",
+    "tslib": "^2.6.2"
+  },
+  "devDependencies": {
+    "typescript": "~5.8.2",
+    "typescript-eslint": "~8.26.0"
+  },
+  "peerDependencies": {
+    "isomorphic-fetch": "^3.0.0"
+  },
+  "engines": {
+    "node": "^20.11.1 || >=22.0.0"
+  }
+}

--- a/packages/@ama-sdk/schematics/testing/mocks/updates/v12/shell-package.results.json
+++ b/packages/@ama-sdk/schematics/testing/mocks/updates/v12/shell-package.results.json
@@ -1,0 +1,53 @@
+{
+  "name": "@o3r-training/showcase-sdk",
+  "version": "0.0.0-placeholder",
+  "description": "showcase-sdk - SDK",
+  "source": "src/index.ts",
+  "main": "cjs/index.js",
+  "module": "esm2020/index.js",
+  "node": "cjs/index.js",
+  "types": "index.d.ts",
+  "sideEffects": false,
+  "deprecated": "This package is intended for testing purposes only.",
+  "publishConfig": {
+    "access": "public"
+  },
+  "exports": {
+    "./package.json": {
+      "default": "./package.json"
+    },
+    ".": {
+      "main": "./cjs/index.js",
+      "module": "./esm2020/index.js",
+      "node": "./cjs/index.js",
+      "default": "./esm2020/index.js",
+      "import": "./esm2020/index.js",
+      "require": "./esm2020/index.js",
+      "types": "./index.d.ts"
+    },
+    "./open-api.yaml": "./open-api.yaml",
+    "./helpers": {
+      "main": "./cjs/helpers/index.js",
+      "node": "./cjs/helpers/index.js",
+      "default": "./esm2020/helpers/index.js",
+      "import": "./esm2020/helpers/index.js",
+      "require": "./esm2020/helpers/index.js",
+      "types": "./helpers/index.d.ts",
+      "module": "./esm2020/helpers/index.js"
+    }
+  },
+  "dependencies": {
+    "@swc/helpers": "~0.5.0",
+    "tslib": "^2.6.2"
+  },
+  "devDependencies": {
+    "typescript": "~5.8.2",
+    "typescript-eslint": "~8.26.0"
+  },
+  "peerDependencies": {
+    "isomorphic-fetch": "^3.0.0"
+  },
+  "engines": {
+    "node": "^20.11.1 || >=22.0.0"
+  }
+}

--- a/packages/@o3r-training/showcase-sdk/package.json
+++ b/packages/@o3r-training/showcase-sdk/package.json
@@ -6,10 +6,6 @@
   "main": "cjs/index.js",
   "module": "esm2020/index.js",
   "node": "cjs/index.js",
-  "es2020": "esm2020/index.js",
-  "esm2020": "esm2020/index.js",
-  "es2015": "esm2015/index.js",
-  "esm2015": "esm2015/index.js",
   "types": "index.d.ts",
   "sideEffects": false,
   "deprecated": "This package is intended for testing purposes only.",
@@ -23,11 +19,9 @@
     ".": {
       "main": "./cjs/index.js",
       "module": "./esm2020/index.js",
-      "es2020": "./esm2020/index.js",
-      "esm2020": "./esm2020/index.js",
       "node": "./cjs/index.js",
       "default": "./esm2020/index.js",
-      "typings": "./index.d.ts",
+      "types": "./index.d.ts",
       "import": "./esm2020/index.js",
       "require": "./esm2020/index.js"
     },

--- a/packages/@o3r-training/training-sdk/package.json
+++ b/packages/@o3r-training/training-sdk/package.json
@@ -7,10 +7,6 @@
   "main": "cjs/index.js",
   "module": "esm2020/index.js",
   "node": "cjs/index.js",
-  "es2020": "esm2020/index.js",
-  "esm2020": "esm2020/index.js",
-  "es2015": "esm2015/index.js",
-  "esm2015": "esm2015/index.js",
   "types": "index.d.ts",
   "sideEffects": false,
   "exports": {
@@ -26,11 +22,9 @@
     ".": {
       "main": "./cjs/index.js",
       "module": "./esm2020/index.js",
-      "es2020": "./esm2020/index.js",
-      "esm2020": "./esm2020/index.js",
       "node": "./cjs/index.js",
       "default": "./esm2020/index.js",
-      "typings": "./index.d.ts",
+      "types": "./index.d.ts",
       "import": "./esm2020/index.js",
       "require": "./esm2020/index.js"
     }

